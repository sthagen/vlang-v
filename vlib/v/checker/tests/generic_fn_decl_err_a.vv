module main

struct Db {}

struct User {}

struct Post {}

fn main() {
	r := Db{}
	u := User{}
	p := Post{}

	r.create(u, p)
}

fn (r Db) create<U>(u U, p P) {
	println('Yo')
}
