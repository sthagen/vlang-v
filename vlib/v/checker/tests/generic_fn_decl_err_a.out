vlib/v/checker/tests/generic_fn_decl_err_a.vv:17:28: error: generic type name `P` is not mentioned in fn `create<U>`
   15 | }
   16 |
   17 | fn (r Db) create<U>(u U, p P) {
      |                            ^
   18 |     println('Yo')
   19 | }
